File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/account/balance/BalanceBuilder.java
package com.fintechplatform.ui.account.balance;

import android.content.Context;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.api.account.balance.api.BalanceAPIModule;
import com.fintechplatform.api.net.NetData;
import com.fintechplatform.api.net.NetModule;


/**
 * Created by tcappellari on 08/12/2017.
 */

public class BalanceBuilder {


    public BalanceHelperComponent createBalanceHelperComponent(String hostName, Context context) {
        return DaggerBalanceHelperComponent.builder()
                .balanceAPIModule(new BalanceAPIModule(hostName))
                .netModule(new NetModule(new NetData(Volley.newRequestQueue(context), hostName)))
                .build();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/account/balance/BalanceBuilder.kt
package com.fintechplatform.ui.account.balance

import android.content.Context
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.account.balance.api.BalanceAPIModule
import com.fintechplatform.api.net.NetModule

class BalanceBuilder {
    fun createBalanceHelperComponent(hostName: String, context: Context): BalanceHelperComponent = 
            DaggerBalanceHelperComponent.builder()
                .balanceAPIModule(BalanceAPIModule(hostName))
                .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                .build()
}